scripts/merge_lora.py

```diff
@@ -18,66 +20,75 @@
+        precision: Optional precision setting to instantiate the model weights in. By default, this will
+            automatically be inferred from the metadata in the given `checkpoint_dir` directory.
     """
     check_valid_checkpoint_dir(checkpoint_dir)
-    out_dir.mkdir(parents=True, exist_ok=True)
+    if pretrained_checkpoint_dir is not None:
+        check_valid_checkpoint_dir(pretrained_checkpoint_dir)
+    if (checkpoint_dir / "lit_model.pth.lora").is_file():
+        print("LoRA weights have already been merged in this checkpoint.")
+        return
 
-    precision = precision or get_default_supported_precision(training=False)
-    fabric = L.Fabric(devices=1, precision=precision)
+    lora_params, pretrained_checkpoint_dir, lora_precision = load_lora_metadata(checkpoint_dir)
+    precision = precision if precision is not None else lora_precision
 
-    config = Config.from_json(
-        checkpoint_dir / "lit_config.json",
-        r=lora_r,
-        alpha=lora_alpha,
-        dropout=lora_dropout,
-        to_query=lora_query,
-        to_key=lora_key,
-        to_value=lora_value,
-        to_projection=lora_projection,
-        to_mlp=lora_mlp,
-        to_head=lora_head,
-    )
+    fabric = L.Fabric(devices=1, precision=precision)
+    config = Config.from_json(checkpoint_dir / "lit_config.json", **lora_params)
 
     with fabric.init_module(empty_init=True):
         model = GPT(config)
-    checkpoint_path = checkpoint_dir / "lit_model.pth"
-    checkpoint = lazy_load(checkpoint_path)
+
+    lora_path = checkpoint_dir / "lit_model.pth"
+    pretrained_checkpoint = lazy_load(pretrained_checkpoint_dir / "lit_model.pth")
     lora_checkpoint = lazy_load(lora_path)
-    checkpoint.update(lora_checkpoint.get("model", lora_checkpoint))
-    model.load_state_dict(checkpoint)
 
+    # Merge LoRA weights into the base model
+    pretrained_checkpoint.update(lora_checkpoint.get("model", lora_checkpoint))
+    model.load_state_dict(pretrained_checkpoint)
     merge_lora_weights(model)
 
-    save_path = out_dir / "lit_model.pth"
-    fabric.print(f"Saving weights to {str(save_path)!r}")
-    # remove lora parameters and the lora linear substring
+    # Remove LoRA parameters and the LoRA linear substring
     state_dict = {k.replace("linear.", ""): v for k, v in model.state_dict().items() if not lora_filter(k, v)}
+    save_path = checkpoint_dir / "lit_model.pth.merged"
     torch.save(state_dict, save_path)
 
+    # Make a backup of the LoRA weights (they are only a few MBs)
+    shutil.move(checkpoint_dir / "lit_model.pth", checkpoint_dir / "lit_model.pth.lora")
+    shutil.move(checkpoint_dir / "lit_model.pth.merged", checkpoint_dir / "lit_model.pth")
+
+    fabric.print(f"Saved merged weights to {str(checkpoint_dir / 'lit_model.pth')!r}")
+    fabric.print(f"A backup of the old LoRA weights is in {str(checkpoint_dir / 'lit_model.pth.lora')!r}")
+
+
+def load_lora_metadata(checkpoint_dir: Path) -> Tuple[Dict[str, Any], Path, str]:
```

Comment:
Technically precision is an Optional[str] since you are using `get`

Reply:
How do you see this stuff 😄

Reply:
I installed mypy into my neurocortex